src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -554,6 +554,7 @@ bool IsInSectionDirectiveBrace()
             // Open brace is a child of the C# code block that is the directive itself
             if (owner is RazorMetaCodeSyntax &&
                 owner.Parent is CSharpCodeBlockSyntax codeBlock &&
+                codeBlock.Children.Count > 3 &&
```

Comment:
Ran into this randomly while debugging, the syntax tree gets weird if there is a compile error immediately before a code block. Probably getting low prism hits on this :)

Reply:
Is this one of the issues we saw reported in last Thursday triage? Sounds similar.

Reply:
Looks like its https://prism.vsdata.io/failure/?query=fe%3Arazor%2Ffault&eventType=fault&failureType=hits&failureHash=9b303b5e-2259-dd32-b0ff-254e89ec5e2c so quite possibly.